### ISSUE TITLE
Allow WhitespaceAround for ARRAY_INIT token

### ISSUE THREAD

**Issue by jsotuyod** [checkstyle/checkstyle#3202]
#Using this configuration:

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAround"/>
    </module>
</module>
```

Checkstyle won't report on the curly braces used for array declarations, such as:

```java
public class Foo {
    public final int[] COLORS = new int[]{Color.BLACK };
    //                                   ^ doesn't report missing whitespace around {
}
```

```bash
$ java -jar ~/Downloads/checkstyle-6.18/checkstyle-6.18-all.jar -c config.xml Foo.java
Starting audit...
Audit done.
```

This is down to the fact that even though `WhitespaceAround` appllies to `LCURLY` and `RCURLY` tokens, when writing an array, the `{` is actually treated as an `ARRAY_INIT` token, which is not currently being processed by the check at all (even is not allowed Token).

It would be desired if `ARRAY_INIT` was to be processed by `WhitespaceAround` in order to allow valiadation of code above, and have the configuration(given before) violation on arrays' initialization.

**Comment by romani** [comment]
please report issue as described at http://checkstyle.sourceforge.net/report_issue.html#How_to_report_a_bug

**Comment by jsotuyod** [comment]
@romani better now?

**Comment by romani** [comment]
please rephrase one more time what behaviour your expect.
do you want to have ability to specify `ARRAY_INIT` in `tokens` property ? (as it is kind of analogue of `RCURLY ` and `LCURLY` ).

**Comment by jsotuyod** [comment]
That is exactly what I would want. Thanks

**Comment by romani** [comment]
I failed to fix is quickly, Check logic need to specially process that token, as token could be multi-line.
Report of failed UT:
```
Expected :..src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java:262:14: '}' is not followed by whitespace.
Actual   :..src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/InputWhitespace.java:255:43: '{' is not followed by whitespace.
````

**Comment by romani** [comment]
Fix is merged
